lib/aws/index.d.ts

```diff
@@ -13,20 +13,22 @@
 
 import { Credentials } from '@aws-sdk/types';
 import Connection from '../Connection';
+import Transport from '../Transport';
 import * as http from 'http';
 import { OpenSearchClientError } from '../errors';
 
-interface AwsSigv4SignerOptions {
+interface AwsV3Sigv4SignerOptions {
   credentials: Credentials;
   region: string;
 }
 
 interface AwsSigv4SignerResponse {
   Connection: Connection;
+  Transport: Transport;
   buildSignedRequestObject(request: any): http.ClientRequestArgs;
 }
 
-declare function AwsSigv4Signer(opts: AwsSigv4SignerOptions): AwsSigv4SignerResponse;
+declare function AwsSigv4Signer(opts: AwsV3Sigv4SignerOptions|any): AwsSigv4SignerResponse;
```

Comment:
Why `any`? Do we not have a type declaration understanding for the aws-sdk V2 style inputs?

Reply:
unfortunately there is not, that's why any.. I've tried to create an `AwsV2Sigv4SignerOptions` that mirror the Config type but it does not like it.. I think a compromise for this would be to add JSDoc and remove the typescript definition for AwsSigv4Signer but your CI will not like that.